### PR TITLE
restrict this to NotFound errors

### PR DESCRIPTION
**Pull request by zouyee**
**What this PR does / why we need it**:
restrict err  to NotFound errors

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: staging/src/k8s.io/client-go/discovery/discovery_client.go [hunk 1]
```diff
@@ -263,8 +262,7 @@ func (d *DiscoveryClient) serverPreferredResources() ([]*metav1.APIResourceList,
 		for _, version := range apiGroup.Versions {
 			groupVersion := schema.GroupVersion{Group: apiGroup.Name, Version: version.Version}
 			apiResourceList, err := d.ServerResourcesForGroupVersion(version.GroupVersion)
-			if err != nil {
-				// TODO: maybe restrict this to NotFound errors
+			if err != nil && errors.IsNotFound(err) {
```
Comment (dixudx) [inline staging/src/k8s.io/client-go/discovery/discovery_client.go]: Same as above

File: staging/src/k8s.io/client-go/discovery/discovery_client.go [hunk 2]
```diff
@@ -200,8 +200,7 @@ func (d *DiscoveryClient) serverResources() ([]*metav1.APIResourceList, error) {
 		for _, version := range apiGroup.Versions {
 			gv := schema.GroupVersion{Group: apiGroup.Name, Version: version.Version}
 			resources, err := d.ServerResourcesForGroupVersion(version.GroupVersion)
-			if err != nil {
-				// TODO: maybe restrict this to NotFound errors
+			if err != nil && errors.IsNotFound(err) {
```
Comment (dixudx) [inline staging/src/k8s.io/client-go/discovery/discovery_client.go]: Simply changing to this is not correct.

What if `err` is not `nil`, but `errors.IsNotFound(err)` is false?

**Review by dixudx — CHANGES_REQUESTED** [review]

**Comment by jsafrane** [comment]
I know very little about client-go
/unassign

**Pull request closed without merging**